src/java/org/apache/cassandra/locator/ReplicaList.java

```diff
@@ -332,4 +330,41 @@ public static ReplicaList toDummyList(Collection<Range<Token>> ranges)
                                .collect(Collectors.toList()));
 
     }
+
+    public void prioritizeForRead()
+    {
+        // TODO: remove in favor of a smarter snitch / replication strategy based approach
+        // put a full replica first
+        int firstFull = -1;
+        for (int i=0; i<size(); i++)
+        {
+            if (get(i).isFull())
+            {
+                firstFull = i;
+                break;
+            }
+        }
+
+        Preconditions.checkState(firstFull >= 0, "At least one full replica required for reads");
+
+        if (firstFull > 0)
+        {
+            Collections.rotate(replicaList.subList(0, firstFull + 1), 1);
```

Comment:
I find the use of rotation a little more confusing then just iterating the collection and selecting into a new collection.

It seems like you select the first full replica, then select transient replicas, then select any remaining required full replicas.